api/src/services/xml_generation/submission_xml_assembler.py

```diff
@@ -0,0 +1,209 @@
+"""Assemble complete submission XML with header, forms, and footer."""
+
+import logging
+from typing import Any
+
+from lxml import etree as lxml_etree
+
+from src.db.models.competition_models import Application, ApplicationSubmission
+from src.services.xml_generation.config import load_xml_transform_config
+from src.services.xml_generation.header_generator import (
+    generate_application_footer_xml,
+    generate_application_header_xml,
+)
+from src.services.xml_generation.models import XMLGenerationRequest
+from src.services.xml_generation.service import XMLGenerationService
+from src.services.xml_generation.utils.attachment_mapping import create_attachment_mapping_from_list
+
+logger = logging.getLogger(__name__)
+
+
+class SubmissionXMLAssembler:
+    """Assembles complete GrantApplication.xml with header, forms, and footer."""
+
+    def __init__(self, application: Application, application_submission: ApplicationSubmission):
+        self.application = application
+        self.application_submission = application_submission
+        self.xml_service = XMLGenerationService()
+
+    def is_form_xml_supported(self, form_name: str) -> bool:
+        """Check if a specific form has XML transformation rules configured."""
+        config = load_xml_transform_config(form_name)
+        return bool(config)
+
+    def get_supported_forms(self) -> list[tuple[Any, str]]:
+        """Get list of application forms that are supported for XML generation.
+
+        Returns:
+            List of tuples (application_form, form_name) for supported forms only
+        """
+        supported_forms = []
+
+        for app_form in self.application.application_forms:
+            form_name = app_form.form.short_form_name
+            if self.is_form_xml_supported(form_name):
+                supported_forms.append((app_form, form_name))
```

Comment:
Instead of cross-referencing a config, can we just do this and then pipe through this json_to_xml_schema?

```suggestion
        for app_form in self.application.application_forms:
            form_name = app_form.form.short_form_name
            if app_form.form.json_to_xml_schema is not None:
                supported_forms.append((app_form, form_name))
```